openpype/hooks/pre_add_last_workfile_arg.py

```diff
@@ -40,5 +42,15 @@ def execute(self):
             self.log.info("Current context does not have any workfile yet.")
             return
 
+        # Determine whether to open workfile post initialization.
+        if self.data["app"].host_name == "maya":
+            project_name = self.data["project_name"]
+            settings = get_project_settings(project_name)
```

Comment:
No need to query project settings. It's readily available using `self.data["project_settings"]`

Reply:
Thanks!